Sources/Action.swift

```diff
@@ -174,6 +174,42 @@ extension Action: ActionProtocol {
 	}
 }
 
+public final class PropertyAction<Output, Error: Swift.Error>: ActionProtocol {
+	public typealias Input = Void
+
+	public let action: Action<Void, Output, Error>
+
+	public var isEnabled: Property<Bool> {
+		return action.isEnabled
+	}
+
+	public init<P: PropertyProtocol, T>(input: P, _ execute: @escaping (T) -> SignalProducer<Output, Error>) where P.Value == T? {
+		action = Action(enabledIf: input.map { $0 != nil }) {
+			execute(input.value!)
+		}
+	}
+
+	public convenience init<P: PropertyProtocol, T>(input: P, _ execute: @escaping (T) -> SignalProducer<Output, Error>) where P.Value == T {
+		self.init(input: input.map(Optional.some), execute)
+	}
+
+	public var events: Signal<Event<Output, Error>, NoError> {
+		return action.events
+	}
+
+	public var values: Signal<Output, NoError> {
+		return action.values
+	}
+
+	public var errors: Signal<Error, NoError> {
+		return action.errors
+	}
+
+	public func apply(_ input: Void = ()) -> SignalProducer<Output, ActionError<Error>> {
```

Comment:
I initially tried to use `func apply() -> SignalProducer` here, but Swift complained about this not matching the signature of `apply` as defined in `ActionProtocol`. The workaround was to at least provide a default value of `()`, but this still feels like a bug to me.